### PR TITLE
Fix sending CATs with the CLI

### PR DESCRIPTION
**Pull request by Quexington**

### CODE REVIEW

**Review by altendky — CHANGES_REQUESTED** [review]
To test this I went ahead and sent some USDS back to Stably.  Other than it not showing up there yet, which they say can take up to a day, it seems to have worked.  I also wrote https://github.com/Chia-Network/chia-blockchain/pull/9942 on top of this PR to add CAT support to `chia wallet get_transaction`.  I should add something to `get_transactions` as well.

Thanks!

File: chia/cmds/wallet.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ def get_transactions_cmd(
 @click.option("-f", "--fingerprint", help="Set the fingerprint to specify which wallet to use", type=int)
 @click.option("-i", "--id", help="Id of the wallet to use", type=int, default=1, show_default=True, required=True)
 @click.option("-a", "--amount", help="How much chia to send, in XCH", type=str, required=True)
+@click.option("-e", "--memo", help="Additional memo for the transaction", type=str, default=None)
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/cmds/wallet.py:91]: If you were using the GUI you'd have the same limited functionality of only one per transaction.  It's sort of a useless parameter anyways I almost didn't add it back.  I'd rather leave it the way it is and discourage excessive use.

File: chia/cmds/wallet_funcs.py [hunk 2]
```diff
@@ -84,17 +84,42 @@ async def send(args: dict, wallet_client: WalletRpcClient, fingerprint: int) ->
...
         return
-    print("Submitting transaction...")
-    final_amount = uint64(int(amount * units["chia"]))
+
+    summaries_response = await wallet_client.get_wallets()
+    final_amount: Optional[uint64] = None
     final_fee = uint64(int(fee * units["chia"]))
-    res = await wallet_client.send_transaction(wallet_id, final_amount, address, final_fee)
+    for summary in summaries_response:
+        if int(wallet_id) == int(summary["id"]):
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/cmds/wallet_funcs.py]: it was but it turns out that the send and get _transactions methods expect a string so after doing this I had to cast those arguments to a string which ended up being more of the same but probably logically better.

File: chia/cmds/wallet_funcs.py [hunk 3]
```diff
@@ -84,17 +84,42 @@ async def send(args: dict, wallet_client: WalletRpcClient, fingerprint: int) ->
...
+            typ: WalletType = WalletType(int(summary["type"]))
+            if typ == WalletType.STANDARD_WALLET:
+                final_amount = uint64(int(amount * units["chia"]))
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/cmds/wallet_funcs.py:107]: `amount` is a `Decimal` and mypy throws an error of `argument 1 to uint64 is unexpected type Decimal` if we don't do this.

File: chia/cmds/wallet.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (altendky) [reply chia/cmds/wallet.py:91]: It also makes for more complicated code dealing with `None`.

File: chia/cmds/wallet_funcs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (altendky) [reply chia/cmds/wallet_funcs.py:107]: Ah, ok.  I'll have to take care of this separately, but we're mis-hinting with `StructStream`.  We also really shouldn't mess with `.__new__()` so much.  The first thing we do is create an `int()` instance which would happen if we just wrote a regular `.__init__()` and then `super()`ed and then we could put our checks after that...

https://github.com/Chia-Network/chia-blockchain/blob/eecc7cd1cd6275712a34e5fadfb6a27ea337a121/chia/util/struct_stream.py#L13

https://github.com/python/typeshed/blob/5d07ebc864577c04366fcc46b84479dbec033921/stdlib/builtins.pyi#L181-L185
```python
class int:
    @overload
    def __new__(cls: type[Self], __x: str | bytes | SupportsInt | SupportsIndex | SupportsTrunc = ...) -> Self: ...
    @overload
    def __new__(cls: type[Self], __x: str | bytes | bytearray, base: SupportsIndex) -> Self: ...
```

**Review by altendky — APPROVED** [review]
Thanks again.  `:]`

**Pull request merged**